### PR TITLE
refactor: Migrate public-page header actions to Vue

### PR DESCRIPTION
**Pull request by susnux**
## Summary

The header menu was still using legacy jQuery UI and duplicated styles in many places instead of reusing the header menu styles we use for the logged-in apps view.

So this PR is migrating that menu to Vue:
1. Make our header actions declarative (adding a `getData` method that allows to just return the data needed for rendering)
2. Keep `render` function for custom entries app developers might have implemented (so this PR is non breaking).
3. Implement the header action menu and add generic entry components
4. Implement special components for the "create federated share" entry and the "direct link" entry.
5. Minor optimization: Add documentation for parameters.

### Screen recording

[Bildschirmaufnahme_20240830_162107.webm](https://github.com/user-attachments/assets/4c2a5fc1-53da-4fd0-8cd5-aacc6e8c272d)

<details>
First version:

[Bildschirmaufnahme_20240828_143026.webm](https://github.com/user-attachments/assets/3d73227d-687e-41b9-98a6-e08840940955)

</details>

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by jancborchardt — DISMISSED** [review]
Nice! :) Some feedback:
- [x] ~~The "Shared by" subline below Nextcloud is missing, or is it not there for a different reason?~~
    ( https://github.com/nextcloud/server/pull/47568#issuecomment-2317209300 )
- [x] ~~The focus effect in the top right includes a small line below the text/icon in the middle, that should not be there (I assume it comes from "active" style from the app navigation for logged in people?)~~
    ( https://github.com/nextcloud/server/pull/47616 )
- [x] The whitespace around the entries in the top right action menu seems a bit large compared to how it should be?
    ( https://github.com/nextcloud-libraries/nextcloud-vue/pull/6026 )

**Comment by skjnldsv** [comment]
> * The "Shared by" subline below Nextcloud is missing, or is it not there for a different reason?

This wasn't touched. The display of the share owner depends on its profile visibility settings :)
That's not relevant to this pr :+1:

**Review by skjnldsv — APPROVED** [review]

**Comment by susnux** [comment]
@jancborchardt all of the remarks are fixed, I updated the screen recording in the initial post to reflect the current state.

File: core/src/components/PublicPageMenu/PublicPageMenuExternalDialog.vue [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+			shareWith: remoteUrl.value,
+			token: getSharingToken(),
+		})
+		if (data.remoteUrl.includes('://')) {
+			window.location.href = data.remoteUrl
+		} else {
+			window.location.href = `${window.location.protocol}//${data.remoteUrl}`
+		}
+	} catch (error) {
+		logger.error(error as Error)
```
Comment (artonge) [inline core/src/components/PublicPageMenu/PublicPageMenuExternalDialog.vue:84]:
```suggestion
		logger.error("Failed to create federated share" { error })
```

**Review by artonge — APPROVED** [review]

**Pull request merged**